### PR TITLE
[OLD] Fix scroll on native when going back to the login page

### PR DESCRIPTION
**Pull request by joh42**
### Details

If the input field is focused when you go back to the login page on native, you are not scrolled to the top of the page. This differs from mweb, where you are scrolled to the top.

### Fixed Issues

$ https://github.com/Expensify/App/issues/23718
PROPOSAL: https://github.com/Expensify/App/issues/23718#issuecomment-1665741165

#### Tests to make sure login is not broken:
1. Go to the login page
2. Tap on the phone/email input field
3. Enter your phone number or email address
4. Click on the button labeled Continue
5. Verify that the magic code input is shown as usual

#### Tests for fixes on native:
With focus
1. Go to the login page
2. Tap on the phone/email input field to open the keyboard
3. Scroll down a bit
4. Exit the app (either by clicking on a link or by clicking on the home button)
5. Reopen the app
6. Verify that you are scrolled back to the top of the login page

Without focus
1. Go to the login page
2. Make sure the phone/email input field is blurred (click outside the input field, the line under it should not be green)
3. Scroll down a bit
4. Exit the app (either by clicking on a link or by clicking on the home button)
5. Reopen the app
6. Verify that you are **not** scrolled back to the top of the login page

- [X] Verify that no errors appear in the JS console

### Offline tests

Same as regular tests

### QA Steps

Same as regular tests

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/138504087/39966539-973a-40b4-a8da-53c6d912ab70

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/138504087/fbe33e0c-cc12-41c6-84e5-239afea9cc05

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/138504087/1abd37b8-6cb3-4ec9-8f6a-171c51b4cf35

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/138504087/9cc450af-02b0-4e8f-9ed4-b24614b2c5cc

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/138504087/ffd342be-29d2-4fdc-a177-c2916491df2d

https://github.com/Expensify/App/assets/138504087/4bd5d807-b450-48cc-aaf9-5718249e0325

https://github.com/Expensify/App/assets/138504087/23019294-6a51-4767-8e3d-65288b5964f2

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/138504087/bdfdc7d0-fa36-4849-a8c3-99842ca29bbf

https://github.com/Expensify/App/assets/138504087/d0c82b4b-9448-40d7-a98b-e6aa5aa6f386

https://github.com/Expensify/App/assets/138504087/dddf14fd-0f1d-4a78-88ae-83fca5da9ca0

</details>

### CODE REVIEW

**Comment by joh42** [comment]
Regarding the failing unit test check, there is a unit test that fails on main. I've [written this on the PR](https://github.com/Expensify/App/pull/19321#issuecomment-1676358882) that I think introduced this, and I have also [reported it as a bug](https://app.slack.com/client/T03SC9DTT/C049HHMV9SM/thread/C049HHMV9SM-1691934002.284919).

**Comment by PauloGasparSv** [comment]
Hey team, do you mind pulling from `main` [to fix the broken test](https://expensify.slack.com/archives/C01GTK53T8Q/p1692018371328929?thread_ts=1692012729.203289&cid=C01GTK53T8Q)?

**Comment by joh42** [comment]
Thanks for the quick fix! No unit tests are failing now, ready for review @mollfpr

**Comment by mollfpr** [comment]
I'm still struggling to run the android app in my local, and it's always failed to build the CMake. Any idea?

**Comment by joh42** [comment]
Hmm no I haven't run into that. If you want I can try to build an APK and send it to you. Alternatively there might be someone on Slack who knows how to solve it, seems like there are a few people who've run into CMake issues.

File: src/pages/signin/LoginForm/index.native.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import BaseLoginForm from './BaseLoginForm';
+import AppStateMonitor from '../../../libs/AppStateMonitor';
+
+const propTypes = {
+    /** Function used to scroll to the top of the page */
+    scrollPageToTop: PropTypes.func,
+};
+const defaultProps = {
+    scrollPageToTop: () => {},
```
Comment (mollfpr) [inline src/pages/signin/LoginForm/index.native.js:11]: Should this be set to `undefined`? So we can early return the listener in the first render?

**Review by mollfpr — CHANGES_REQUESTED** [review]
Sorry for the delay. I manage to run the Android app. Overall the code and test look good, just a couple of comments.

File: src/pages/signin/LoginForm/index.native.js (hunk 1 above)
Comment (joh42) [reply src/pages/signin/LoginForm/index.native.js]: Makes sense, updated

File: src/pages/signin/LoginForm/index.native.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+};
+
+function LoginForm(props) {
+    const loginFormRef = useRef();
+    const {scrollPageToTop} = props;
```
[the remark this answers is not in the document]
Comment (joh42) [reply src/pages/signin/LoginForm/index.native.js:16]: We still need the props object so we can pass it to the BaseLoginForm:

https://github.com/joh42/expensify-app/blob/6a861a0ff32af9e6b0a4b3febe455808cce95c70/src/pages/signin/LoginForm/index.native.js#L36-L42

But if we still want to destructure there something like `function LoginForm({ scrollPageToTop, ...props })` could work

**Comment by joh42** [comment]
Updated code/responded to comment @mollfpr

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

File: src/pages/signin/LoginForm/index.native.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mollfpr) [reply src/pages/signin/LoginForm/index.native.js:16]: Sorry, I didn't see `props` spread to the component. Let's keep it then.

File: src/pages/signin/LoginForm/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import BaseLoginForm from './BaseLoginForm';
+
+const propTypes = {
+    /** Function used to scroll to the top of the page */
+    scrollPageToTop: PropTypes.func,
+};
+const defaultProps = {
+    scrollPageToTop: () => {},
```
Comment (mollfpr) [inline src/pages/signin/LoginForm/index.js:10]: Let's update this to `undefined` to get aligned with the others.

Comment (joh42) [reply src/pages/signin/LoginForm/index.js]: 👍 updated @mollfpr

**Comment by joh42** [comment]
Friendly ping @mollfpr, would be great if we can get this merged within 9 days of assignment

**Comment by mollfpr** [comment]
The code looks good and tests well. @joh42 Could you resolve the conflict? Thanks!

**Comment by joh42** [comment]
Sorry about that, resolved now @mollfpr

**Comment by joh42** [comment]
Welp... so while I did resolve the conflict, the commit wasn't signed, and I didn't notice until after I had pushed. I know you're not supposed to force push when a PR review has been started, but reverting won't remove the unsigned commit. Is it okay if I remove the unsigned commit and force push in this scenario @mollfpr?

**Comment by joh42** [comment]
While that did allow me to sign the commit locally, I'm still unable to push without using force since my local branch differs from the one on Github @mollfpr.

Since there are new conflicts, I think the cleanest solution might be to remove the unsigned conflict resolution commit and instead resolve all conflicts in a signed commit. But again, that would require a force push.

**Comment by mollfpr** [comment]
@joh42 Okay, let's open a new PR to get cleaner commit and seems everything looks good.

**Pull request closed without merging**